### ISSUE TITLE
Bug: [triple-slash-reference] must not report an error if sourceType "script"

### ISSUE THREAD

**Issue by kkmuffme** [typescript-eslint/typescript-eslint#11837]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

configuration flag

### Repro Code

```TypeScript
configuration flag
```

### ESLint Config

```javascript
https://eslint.org/docs/latest/use/configure/language-options
`sourceType: "script"`
```

### tsconfig

```jsonc

```

### Expected Result

No error for https://typescript-eslint.io/rules/triple-slash-reference/

### Actual Result

Error

### Additional Info

sourceType "script" is non-module, which means "import" (or "require") cannot be used, therefore triple-slash directives must be usable without error

**Comment by bradzacher** [comment]
Triple slash references are still completely fine to ban in "script" files - for example commonjs requires still work in those files, or you can use inline type imports as well (`type T = import('module')`).

It is up to you to configure your repo so that the rule does not run on files you do not want it to.

If you have a file that you want to allow triple slash references in - you must disable the rule for that file.

**Comment by bradzacher** [comment]
Triple slash references are still completely fine to ban in "script" files - for example commonjs requires still work in those files, or you can use inline type imports as well (`type T = import('module')`).

It is up to you to configure your repo so that the rule does not run on files you do not want it to.

If you have a file that you want to allow triple slash references in - you must disable the rule for that file.

**Comment by kkmuffme** [comment]
>for example commonjs requires still work in those files

No they don't - require() only works in node.js and eslint (since flat config) specifically has sourceType "commonjs" for that, which is separate from "script"

>can use inline type imports as well (type T = import('module')).

No, this only works in "module" - which is sourceType "module"

In eslint sourceType "script" means a plain old browser script.
